src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -400,234 +443,223 @@ End Class
+            Dim buffer As Byte() = New Byte(s.Length - 1) {{}}
+            Await {0}
         End Using
     End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 53));
-        }
-
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_Inline()
-        {
-            return AnalyzeVBAsync(@"
+End Module
+            ";
+            return GetVBReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 9, 19, 9, endColumn);
+        }
+
+        [Theory]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length)",
+                    false, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length)",
+                    true, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    false, 95)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    true, 95)]
+        public Task VB_Analyzer_Diagnostic_InlineByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 Imports System
 Imports System.IO
 Imports System.Threading
-Class C
+Public Module C
     Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Await fs.ReadAsync(New Byte(fs.Length - 1) {}, 0, fs.Length)
+        Using s As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Await {0}
         End Using
     End Sub
-End Class
-            ", GetVBResult(8, 19, 8, 73));
+End Module
+            ";
+            return GetVBReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 8, 19, 8, endColumn);
         }
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_CancellationToken()
-        {
-            return AnalyzeVBAsync(@"
-Imports System
-Imports System.IO
-Imports System.Threading
-Class C
-    Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
-            Await fs.ReadAsync(buffer, 0, fs.Length, New CancellationToken())
-        End Using
-    End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 78));
-        }
+        #endregion
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_ConfigureAwait()
+        #region Helpers
+
+        private const string AsyncMethodName = "Read";
+
+        private Task GetCSReadAsyncDiagnostic(string source, string originalArgs, string fixedArgs, bool withConfigureAwait, int startLine, int startColumn, int endLine, int endColumn)
```

Comment:
This naming seems weird as it is not getting diagnostic, it is actually verifying the code fix. We should rename this to something like say `CSharpVerifyCodeFixAsync`